src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -1251,15 +1252,17 @@ protected String checkJavaVersion(final PrintStream logger, String javaCommand,
                         getTimestamp(), javaCommand, versionStr));
 
                 // parse as a number and we should be OK as all we care about is up through the first dot.
+                final VersionNumber minJavaLevel = JavaProvider.getMinJavaLevel();
                 try {
                     final Number version =
                         NumberFormat.getNumberInstance(Locale.US).parse(versionStr);
-                    if(version.doubleValue() < 1.5) {
+                    //TODO: burn it with fire
+                    if(version.doubleValue() < Double.parseDouble("1."+minJavaLevel)) {
```

Comment:
Yes, we could. Such approach is better then the original one I'd guess. Will need to write tests though